### DISCUSSION TITLE
"Avoiding recreating the ref contents" disallowed by compiler lint

### DISCUSSION THREAD

**Discussion by yepitschunked** [reactwg/react-compiler#20] [Something went wrong]
The ref docs have a section on avoiding recreating the ref contents by doing something like this: https://react.dev/reference/react/useRef#avoiding-recreating-the-ref-contents

```
function Video() {
  const playerRef = useRef(null);
  if (playerRef.current === null) {
    playerRef.current = new VideoPlayer();
  }
```

This fails the react compiler eslint check with `Ref values (the `current` property) may not be accessed during render. (https://react.dev/reference/react/useRef)`. Is this intentional?

**Comment by mvitousek** [comment]
Sorry it took us so long to address this! The compiler previously wasn't able to support this pattern because it didn't recognize a distinction between safely avoiding recreating the ref contents and arbitrary, unsafe reading and writing of refs. We just landed https://github.com/facebook/react/pull/31188, which should address this issue and allow this pattern to work without signaling an error or bailing out the compiler.
